src/components/Attachment/Attachment.tsx

```diff
@@ -9,30 +9,30 @@ import {
   ChildrenComponentProps,
   commonPropTypes,
 } from '../../lib'
-import { Extendable, ShorthandRenderFunction, ShorthandValue } from '../../../types/utils'
+import { Extendable, Nullable, ShorthandRenderFunction, ShorthandValue } from '../../../types/utils'
 import Icon from '../Icon/Icon'
 import Button from '../Button/Button'
 import Text from '../Text/Text'
 import Slot from '../Slot/Slot'
 
 export interface AttachmentProps extends UIComponentProps, ChildrenComponentProps {
   /** Button shorthand for the action slot. */
-  action?: ShorthandValue
+  action?: Nullable<ShorthandValue>
```

Comment:
maybe it would be easier to use generic type that will be able to transform entire props type, like that:

```jsx
export interface AttachmentProps {
  action?: ShorthandValue,
  ...
}

class Attachment extends UIComponent<Nullable<AttachmentProps>, ...>
```